potodo/potodo.py

```diff
@@ -217,9 +120,92 @@ def exec_potodo(
                     above,
                     below,
                     counts,
+                    json_format
                 )
+
         # Once all files have been processed, print the dir and the files
-        print_dir_stats(directory_name, buffer, folder_stats, printed_list)
+        # or store them into a dict to print them once all directories have
+        # been processed.
+        if json_format:
+            add_dir_stats(directory_name, buffer, folder_stats, printed_list, dir_stats)
+        else:
+            print_dir_stats(directory_name, buffer, folder_stats, printed_list)
+
+    if json_format:
+        print(json.dumps(dir_stats, indent=4, separators=(',', ': '), sort_keys=False))
+
+
+def buffer_add(
+    buffer: list,
+    folder_stats: List[int],
+    printed_list: List[bool],
+    po_file_stats: PoFileStats,
+    issue_reservations: Mapping[str, str],
+    above: int,
+    below: int,
+    counts: bool,
+    json_format: bool,
+) -> None:
+    """Will add to the buffer the information to print about the file is
+    the file isn't translated entirely or above or below requested
+    values.
+    """
+    # If the file is completely translated,
+    # or is translated below what's requested
+    # or is translated above what's requested
+    if po_file_stats.percent_translated == 100 or \
+            po_file_stats.percent_translated < above or \
+            po_file_stats.percent_translated > below:
+
+        # add the percentage of the file to the stats of the folder
+        folder_stats.append(po_file_stats.percent_translated)
+
+        if not json_format:
+            # don't print that file
+            printed_list.append(False)
+
+        # return without adding anything to the buffer
+        return
+
+    # nb of fuzzies in the file IF there are some fuzzies in the file
+    fuzzy_nb = po_file_stats.fuzzy_nb if po_file_stats.fuzzy_entries else 0
+    # number of entries translated
+    translated_nb = po_file_stats.translated_nb
+    # file size
+    po_file_size = po_file_stats.po_file_size
+    # percentage of the file already translated
+    percent_translated = po_file_stats.percent_translated
+    # `reserved by` if the file is reserved unless the offline/hide_reservation are enabled
+    reserved_by = issue_reservations.get(po_file_stats.filename_dir.lower(), None)
+
+    if json_format:
+        # the order of the keys is the display order
+        desc = dict(name=f"{po_file_stats.directory}/{po_file_stats.filename.strip('.po')}",
```

Comment:
Je trouve ca bizzare le nom et le path, je sais pas trop comment j'aurais fait autrement, mais y'a pas mal de possibilités

`filename`: `xxx.po`
`filepath`: `folder/file.po`
`path`: `relative/path/to/file.po`
`path`: `/absolute/path/to/file.po`
et d'autres

Je sais pas ce qui est le mieux

Reply:
J'ai mis « name » pour le nom « compréhensible par un humain » (le dossier + le nom du fichier sans extension), et « path » pour le côté « script ».
Dans l'idée, l'output JSON est surtout un moyen de bâtir des choses au-dessus de potodo — en premier lieu des tests, d'où l'idée d'avoir ces deux infos. Pas enthousiaste à l'idée de multiplier les champs par contre (pas besoin de chemin relatif, ni de nom de fichier sans extension, ça se déduit des autres).

Reply:
Ca me va !